etna/generic_pages/models.py

```diff
@@ -1,17 +1,28 @@
 from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.db import models
 from django.utils.translation import gettext_lazy as _
 
-from wagtail.admin.panels import FieldPanel
+from modelcluster.fields import ParentalKey
+from wagtail.admin.panels import FieldPanel, InlinePanel, MultiFieldPanel
 from wagtail.api import APIField
 from wagtail.fields import RichTextField, StreamField
+from wagtail.images import get_image_model_string
+from wagtail.models import Orderable
 
-from etna.core.models import BasePage
-from etna.core.serializers import RichTextSerializer
+from rest_framework import serializers
+
+from etna.core.models import BasePageWithNonRequiredIntro, HeroImageMixin
+from etna.core.serializers import (
+    DefaultPageSerializer,
+    ImageSerializer,
+    RichTextSerializer,
+)
 
 from .blocks import GeneralPageStreamBlock
 
 
-class GeneralPage(BasePage):
+class GeneralPage(BasePageWithNonRequiredIntro):
```

Comment:
This feels...awkward 🤔

How difficult would it be to change the `BasePage` to not have a required introduction and then have the separate model for one that does require it?

Reply:
Okay, that sounds fair. I don't think I looked enough into the `BasePage` class.

Those two classes are a lot more readable than `BasePageWithNonRequiredIntro` 👍🏼